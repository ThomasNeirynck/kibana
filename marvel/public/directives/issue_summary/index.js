define(function (require) {
  var module = require('ui/modules').get('monitoring/directives', []);
  var template = require('plugins/monitoring/directives/issue_summary/index.html');

  module.directive('monitoringIssueSummary', function () {
    return {
      restrict: 'E',
      scope: {
        title: '@',
        source: '='
      },
      template: template
    };
  });
});
