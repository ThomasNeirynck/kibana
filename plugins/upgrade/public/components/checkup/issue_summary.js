import PropTypes from 'prop-types';
import React from 'react';

import { DEPRECATION_ISSUE_LEVELS } from '../../lib/constants';


const ISSUE_LEVEL_MESSAGES = {
  info: (count) => (
    <span className="kuiStatusText kuiStatusText--info">
      <span className="kuiStatusText__icon kuiIcon fa-info" />
      {
        count === 1
        ? <span>{ count } note</span>
        : <span>{ count } notes</span>
      }
    </span>
  ),
  warning: (count) => (
    <span className="kuiStatusText kuiStatusText--warning">
      <span className="kuiStatusText__icon kuiIcon fa-bolt" />
      {
        count === 1
        ? <span>{ count } warning</span>
        : <span>{ count } warnings</span>
      }
    </span>
  ),
  critical: (count) => (
    <span className="kuiStatusText kuiStatusText--error">
      <span className="kuiStatusText__icon kuiIcon fa-warning" />
      {
        count === 1
        ? <span>{ count } error</span>
        : <span>{ count } errors</span>
      }
    </span>
  ),
  none: () => (
    <span className="kuiStatusText kuiStatusText--success">
      <span className="kuiStatusText__icon kuiIcon fa-check" />
      No problems
    </span>
  ),
};

export function IssueSummary({ className, issueLevelCounts }) {
  const highestIssueLevel = DEPRECATION_ISSUE_LEVELS.reduce((highestLevel, issueLevel) => {
    if (highestLevel !== 'none') {
      return highestLevel;
    }

    if ((issueLevelCounts[issueLevel] || 0) > 0) {
      return issueLevel;
    } else {
      return highestLevel;
    }
  }, 'none');

  return (
    <span className={className}>
      { ISSUE_LEVEL_MESSAGES[highestIssueLevel](issueLevelCounts[highestIssueLevel]) }
    </span>
  );
}

IssueSummary.propTypes = {
  className: PropTypes.string,
  issueLevelCounts: PropTypes.object,
};

IssueSummary.defaultProps = {
  className: null,
  issueLevelCounts: {},
};
