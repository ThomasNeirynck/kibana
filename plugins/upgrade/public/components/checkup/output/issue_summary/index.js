export { IssueSummary } from './issue_summary';
