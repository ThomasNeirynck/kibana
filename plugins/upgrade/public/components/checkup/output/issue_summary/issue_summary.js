import React from 'react';

import { DEPRECATION_ISSUE_LEVELS } from '../../../../lib/constants';


const ISSUE_LEVEL_MESSAGES = {
  info: (count) => (
    <span className="kuiStatusText kuiStatusText--info">
      <span className="kuiStatusText__icon kuiIcon fa-info" />
      {
        count === 1
        ? <span>{ count } note found</span>
        : <span>{ count } notes found</span>
      }
    </span>
  ),
  warning: (count) => (
    <span className="kuiStatusText kuiStatusText--warning">
      <span className="kuiStatusText__icon kuiIcon fa-bolt" />
      {
        count === 1
        ? <span>{ count } warning found</span>
        : <span>{ count } warnings found</span>
      }
    </span>
  ),
  critical: (count) => (
    <span className="kuiStatusText kuiStatusText--error">
      <span className="kuiStatusText__icon kuiIcon fa-warning" />
      {
        count === 1
        ? <span>{ count } error found</span>
        : <span>{ count } errors found</span>
      }
    </span>
  ),
  none: () => (
    <span className="kuiStatusText kuiStatusText--success">
      <span className="kuiStatusText__icon kuiIcon fa-check" />
      No problems found
    </span>
  ),
};

export function IssueSummary({ className, issueLevelCounts }) {
  const highestIssueLevel = DEPRECATION_ISSUE_LEVELS.reduce((highestLevel, issueLevel) => {
    if (highestLevel !== 'none') {
      return highestLevel;
    }

    if ((issueLevelCounts[issueLevel] || 0) > 0) {
      return issueLevel;
    } else {
      return highestLevel;
    }
  }, 'none');

  return (
    <span className={ className }>
      { ISSUE_LEVEL_MESSAGES[highestIssueLevel](issueLevelCounts[highestIssueLevel]) }
    </span>
  );
}

IssueSummary.propTypes = {
  className: React.PropTypes.string,
  issueLevelCounts: React.PropTypes.object,
};

IssueSummary.defaultProps = {
  className: null,
  issueLevelCounts: {},
};
