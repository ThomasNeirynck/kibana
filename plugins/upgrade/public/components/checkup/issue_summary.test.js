import { shallow } from 'enzyme';
import React from 'react';

import { IssueSummary } from './issue_summary';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('IssueSummary', () => {
  test('renders the className property', () => {
    const component = (
      <IssueSummary className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a success message when all issue level counts are 0', () => {
    const issueLevelCounts = {
      'critical': 0,
      'info': 0,
      'none': 0,
      'warning': 0,
    };
    const component = (
      <IssueSummary issueLevelCounts={issueLevelCounts} />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a success message when only the none level count is non-zero', () => {
    const issueLevelCounts = {
      'critical': 0,
      'info': 0,
      'none': 4,
      'warning': 0,
    };
    const component = (
      <IssueSummary issueLevelCounts={issueLevelCounts} />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders an info message when only none and info level counts are non-zero', () => {
    const issueLevelCounts = {
      'critical': 0,
      'info': 3,
      'none': 4,
      'warning': 0,
    };
    const component = (
      <IssueSummary issueLevelCounts={issueLevelCounts} />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a warning message when only none, info and warning level counts are non-zero', () => {
    const issueLevelCounts = {
      'critical': 0,
      'info': 3,
      'none': 4,
      'warning': 2,
    };
    const component = (
      <IssueSummary issueLevelCounts={issueLevelCounts} />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a critical message when the critical level count is non-zero', () => {
    const issueLevelCounts = {
      'critical': 1,
      'info': 3,
      'none': 4,
      'warning': 2,
    };
    const component = (
      <IssueSummary issueLevelCounts={issueLevelCounts} />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
