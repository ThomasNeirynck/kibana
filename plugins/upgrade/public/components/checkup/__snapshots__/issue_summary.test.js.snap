// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSummary renders a critical message when the critical level count is non-zero 1`] = `
<span
  className={null}
>
  <span
    className="kuiStatusText kuiStatusText--error"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-warning"
    />
    <span>
      1
       error
    </span>
  </span>
</span>
`;

exports[`IssueSummary renders a success message when all issue level counts are 0 1`] = `
<span
  className={null}
>
  <span
    className="kuiStatusText kuiStatusText--success"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-check"
    />
    No problems
  </span>
</span>
`;

exports[`IssueSummary renders a success message when only the none level count is non-zero 1`] = `
<span
  className={null}
>
  <span
    className="kuiStatusText kuiStatusText--success"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-check"
    />
    No problems
  </span>
</span>
`;

exports[`IssueSummary renders a warning message when only none, info and warning level counts are non-zero 1`] = `
<span
  className={null}
>
  <span
    className="kuiStatusText kuiStatusText--warning"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-bolt"
    />
    <span>
      2
       warnings
    </span>
  </span>
</span>
`;

exports[`IssueSummary renders an info message when only none and info level counts are non-zero 1`] = `
<span
  className={null}
>
  <span
    className="kuiStatusText kuiStatusText--info"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-info"
    />
    <span>
      3
       notes
    </span>
  </span>
</span>
`;

exports[`IssueSummary renders the className property 1`] = `
<span
  className="myClass"
>
  <span
    className="kuiStatusText kuiStatusText--success"
  >
    <span
      className="kuiStatusText__icon kuiIcon fa-check"
    />
    No problems
  </span>
</span>
`;
